explore/src/main/scala/explore/constraints/ConstraintsPanel.scala

```diff
@@ -42,30 +42,23 @@ import react.semanticui.collections.form.Form
 import react.semanticui.elements.label.LabelPointing
 
 final case class ConstraintsPanel(
-  obsIds:           List[Observation.Id],
-  constraintSet:    View[ConstraintSet],
-  undoStacks:       View[UndoStacks[IO, ConstraintSet]],
-  renderInTitle:    Tile.RenderInTitle
-)(implicit val ctx: AppContextIO)
+  obsIds:        List[Observation.Id],
+  constraintSet: View[ConstraintSet],
+  undoStacks:    View[UndoStacks[IO, ConstraintSet]],
+  renderInTitle: Tile.RenderInTitle
+)(using val ctx: AppContextIO)
     extends ReactFnProps[ConstraintsPanel](ConstraintsPanel.component)
 
 object ConstraintsPanel {
-  protected type Props = ConstraintsPanel
+  type Props = ConstraintsPanel
 
-  private sealed abstract class ElevationRangeType(val label: String)
-      extends Product
-      with Serializable
+  private enum ElevationRangeType(val label: String):
+    case AirMass   extends ElevationRangeType("Air Mass")
+    case HourAngle extends ElevationRangeType("Hour Angle")
 
-  private object ElevationRangeType {
-    case object AirMass   extends ElevationRangeType("Air Mass")
-    case object HourAngle extends ElevationRangeType("Hour Angle")
-
-    implicit val enumeratedElevationRangeType: Enumerated[ElevationRangeType] =
-      Enumerated.of(AirMass, HourAngle)
-
-    implicit val displayElevationRangeType: Display[ElevationRangeType] =
-      Display.byShortName(_.label)
-  }
+  private object ElevationRangeType:
+    given Enumerated[ElevationRangeType] = Enumerated.from(AirMass, HourAngle).withTag(_.label)
```

Comment:
Maybe this one can be derived too?

Reply:
Yes, there may be others too. The goal here wasn't to migrate everything yet, just to get things to compile :D

Reply:
Oh wait, the derivation macro doesn't work in nested `enum`s. I couldn't figure it out, but there's probably a way.

In these cases we can either move the `enum` out of wherever it's nested, or manually declare the `Enumerated`.

Reply:
Let's do a separate PR changing the remaining `Enumerated` declarations to `enum` + `derives`.